keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -24,21 +26,52 @@ import java.time.Instant
 
 class SqlArtifactRepository(
   private val jooq: DSLContext,
-  private val clock: Clock
+  private val clock: Clock,
+  private val objectMapper: ObjectMapper
 ) : ArtifactRepository {
   override fun register(artifact: DeliveryArtifact) {
     jooq.insertInto(DELIVERY_ARTIFACT)
       .set(DELIVERY_ARTIFACT.UID, randomUID().toString())
       .set(DELIVERY_ARTIFACT.NAME, artifact.name)
       .set(DELIVERY_ARTIFACT.TYPE, artifact.type.name)
-      .onDuplicateKeyIgnore()
+      .set(DELIVERY_ARTIFACT.DETAILS, artifact.toJson())
+      .onDuplicateKeyUpdate()
+      .set(DELIVERY_ARTIFACT.DETAILS, artifact.toJson())
       .execute()
-      .also { count ->
-        if (count == 0) log.warn("Duplicate artifact registered: {}", artifact)
-      }
   }
 
-  override fun store(artifact: DeliveryArtifact, version: String, status: ArtifactStatus): Boolean {
+  private fun DeliveryArtifact.toJson() =
```

Comment:
Maybe `detailsAsJson()` since this only returns the details, not all properties of the artifact?